src/libraries/System.Runtime.Numerics/src/System/Numerics/BigNumber.cs

```diff
@@ -511,6 +515,96 @@ private static ParsingStatus HexNumberToBigInteger(ref BigNumberBuffer number, o
             }
         }
 
+        private static ParsingStatus BinNumberToBigInteger(ref BigNumberBuffer number, out BigInteger result)
+        {
+            if (number.digits is null || number.digits.Length < 2)
+            {
+                result = default;
+                return ParsingStatus.Failed;
+            }
+
+            const int DigitsPerBlock = 32;
+
+            int totalDigitCount = number.digits.Length - 1;   // Ignore trailing '\0'
+
+            int blockCount = Math.DivRem(totalDigitCount, DigitsPerBlock, out int remainingDigitsInBlock);
```

Comment:
nit: We should prefer `(int blockCount, int remainingDigitsInBlock) = int.DivRem(totalDigitCount, DigitsPerBlock)` in new code

Reply:
Fixed.